src/newsfeed/infrastructure/event_storages.py

```diff
@@ -84,6 +92,108 @@ def __init__(self, config: Dict[str, str]):
             del newsfeed_storage[event_index]
 
 
+class RedisEventStorage(EventStorage):
+    """Event storage that stores events in redis."""
+
+    def __init__(self, config: Dict[str, str]):
+        """Initialize storage."""
+        super().__init__(config)
+
+        redis_config = parse_redis_dsn(config['dsn'])
+        self._pool = aioredis.pool.ConnectionsPool(
+            address=redis_config['address'],
+            db=int(redis_config['db']),
+            create_connection_timeout=int(redis_config['connection_timeout']),
+            minsize=int(redis_config['minsize']),
+            maxsize=int(redis_config['maxsize']),
+            encoding='utf-8',
+        )
+        self._max_newsfeed_ids = int(config['max_newsfeeds'])
+        self._max_events_per_newsfeed_id = int(config['max_events_per_newsfeed'])
+
+    async def get_by_newsfeed_id(self, newsfeed_id: str) -> Iterable[EventData]:
+        """Get events data from storage."""
+        async with self._get_connection() as redis:
+            newsfeed_storage = []
+            # TODO:
+            #  - Check async
+            #  - Write base serializer
+            #  - Is it correct place for ordering?
+            #  - Ordering by `published_at`?
+            async for key, _ in redis.izscan(f'newsfeed_id:{newsfeed_id}'):
```

Comment:
Please, take a look on how we can use lists here so that complexity of that select is O(1).

Reply:
If we'll use a list, we'll read the full list with O(n) complexity as well,